app/settings/surveys/survey-editor.directive.js

```diff
@@ -637,7 +637,7 @@ function SurveyEditorController(
         .saveCache(_.extend({ roles: $scope.roles_allowed }, { formId: $scope.survey.id }))
         .$promise
         .then(function (roles) {
-            $location.path('settings/surveys/edit/' + $scope.survey.id);
+            $location.url('/settings/surveys');
```

Comment:
This makes sense. Theres an existing issue with both this and #676, that we're not waiting for *all* the requests to finish before redirecting, ie. roles might finish saving, but attributes + tasks are still processing, yet we we redirect anyway.

Do you think you could return promises from `saveRoles` `saveTasks` and `saveAttributes` then handle the redirect in `saveSurvey` ?
That enables catching any errors during save and only redirecting if *all* of them succeed.